app/services/delete_content_item.rb

```diff
@@ -0,0 +1,40 @@
+module Services
+  class DeleteContentItem
+    def self.destroy_content_items_with_links(content_ids)
+      content_ids = Array(content_ids)
+
+      content_items = ContentItem.where(content_id: content_ids)
+      destroy_supporting_objects(content_items)
+      content_items.destroy_all
+
+      destroy_links(content_ids)
+    end
+
+    def self.destroy_supporting_objects(content_items)
+      content_items = Array(content_items)
+
+      supporting_classes = [
+        AccessLimit,
+        Location,
+        State,
+        Translation,
+        Unpublishing,
+        UserFacingVersion,
+        ChangeNote,
+      ]
+
+      supporting_classes.each do |klass|
+        next unless ActiveRecord::Base.connection.data_source_exists?(klass.table_name)
+        klass.where(content_item: content_items).destroy_all
+      end
+
+      LockVersion.where(target: content_items).destroy_all
+    end
+
+    def self.destroy_links(content_ids)
+      content_ids = Array(content_ids)
+      LinkSet.where(content_id: content_ids).destroy_all
+      Link.where(target_content_id: content_ids).destroy_all
```

Comment:
This will delete links from other content to this content, which might not be the desired thing. If we're replacing eg. a Publication and deleting all links TO it, that Publication will be removed from any Document Collections that were linking to it. I think only removing links FROM this content is required.